### ISSUE TITLE
Official integration with HA

### ISSUE THREAD

**Issue by ausil** [firstof9/openevse#245]
Given how old and that it is fully manual in its configuration the existing OpenEVSE support in HA should be replaced with this integration. What do you think is remaining for it to be ready?  I think it is ready today. I am happy to help get the integration merged.

**Comment by firstof9** [comment]
I tried to take over as codeowner for it in HA, no one ever reviewed the PR

**Comment by Oinq** [comment]
It's a pity I can't code.
I'm pondering the fact of buying a commercial bidirectional EVSE just
because I can't connect this easily with HA...


-----

Cumprimentos   /   Kind Regards  /  Met vriendelijke groeten / Sincères
amitiés
Nuno Leitão
https://www.linkedin.com/in/nunocasanovaleitao/

**Comment by firstof9** [comment]
This integration works great with HA, just load it in via HACS for easy mode.

**Comment by Oinq** [comment]
can't find it in hacs, any pointer?

[image: image.png]


-----

Cumprimentos   /   Kind Regards  /  Met vriendelijke groeten / Sincères
amitiés
Nuno Leitão
https://www.linkedin.com/in/nunocasanovaleitao/

**Comment by firstof9** [comment]
Add this repo to HACS via "Custom Repositories"

**Comment by jamesb2147** [comment]
Can you point me to where it's languishing? I'm not seeing any outstanding PR's in the HA Github relating to OpenEVSE except that somebody wants to rewrite it to MQTT as a breaking change, and I can see that's not you.

I'm no code expert, but I can be "persistent". :) I did manage to get some docs updates merged a few months back, and would happily cheerlead/shepherd a PR to transfer ownership.

**Comment by jamesb2147** [comment]
Think you can add support for importing the existing integration? Not breaking the existing users' config makes it a lot easier of a pill to swallow for accepting the merge request.

**Comment by firstof9** [comment]
First steps for this were here:

https://github.com/home-assistant/core/pull/55303

**Comment by fhteagle** [comment]
If there's somewhere we can chime in to support your bid to supplant the old/official integration, please let us know. Your version has been rock solid for ~6 months now...

**Comment by firstof9** [comment]
I have no idea, maybe make some noise on discord?

**Comment by darrendavid** [comment]
I've got the OpenEVSE integration installed from HACS and I've added in my PV system information into the configuration, but it doesn't seem to be limiting the charge rate based on what I'm feeding back to the grid. The logs are showing the following every second:

`2024-01-24 13:48:17.661 INFO (MainThread) [custom_components.openevse] Could not update status for available_current
2024-01-24 13:48:17.661 INFO (MainThread) [custom_components.openevse] Could not update status for smoothed_available_current`

How might I troubleshoot?

**Comment by firstof9** [comment]
> How might I troubleshoot?

Please create a new issue, but you can enable debugging by clicking the "enable debugging" button from the integration screen.
![image](https://github.com/firstof9/openevse/assets/1105672/241954b8-dd2c-4a90-8738-576f74c0a8bf)

This will start debug logging. Let it run for a minute or two then press stop, and it'll download your HA logfile and attach it to your new issue.

**Comment by jamesb2147** [comment]
FYI I asked in Discord the other day and noted that the official number of users of the existing integration was 0...

I think if you opened a new merge request to get pulled into HA proper, it'd get approved this time. I'd certainly go to bat for you on replacing the existing integration, and I suspect several others would as well.

**Comment by firstof9** [comment]
My only concern with that is far less ability to get things fixed quickly.

**Comment by jamesb2147** [comment]
I mean, you'd know better than I, but isn't the interface fairly stable? I'd imagine the primary stuff that'd need fast iteration is new features, and those are exactly the type of thing that can wait for a monthly release (which isn't a bad cadence!).

Actual bugfixes are released fairly frequently (e.g. 2x a week) after a monthly release.

**Comment by pdcastro** [comment]
> _My only concern with that is far less ability to get things fixed quickly._

On the other hand:

* You’d get much increased visibility and more users, and presumably more contributions. The whole OpenEVSE project would get greater visibility. A few hours ago I started a web search for EV chargers that offered a local API and a Home Assistant integration (to buy). Soon enough I came across OpenEVSE, but their official HASS integration put me off, and I did not know about any “unofficial” integration. The [emonCMS stuff](https://docs.openenergymonitor.org/emoncms/index.html) looks focused on monitoring, not control. Then I started looking into other projects and eventually, by chance, I came across a link to this repo. This integration looks like a proper control interface, usable through HASS automations and scripts.

* Presumably you could keep this custom integration going in parallel as well. If in forums / issues / discussions people were left waiting because the official HASS process was slow (not saying that it is), you could point them to the custom integration. In this sense, you could treat the official HASS integration as free advertising. 🙂

> _Actual bugfixes are released fairly frequently (e.g. 2x a week) after a monthly release._

Right? I would not call that slow. Even once a week would be fine.
